src/modules/products/products.service.ts

```diff
@@ -62,4 +56,29 @@ export class ProductsService {
     if (quantity === 0) return ProductStatusType.OUT_STOCK;
     return quantity >= 5 ? ProductStatusType.IN_STOCK : ProductStatusType.LOW_STOCK;
   }
+
+  async deleteProduct(productId: string, orgId: string) {
+    const product = await this.productRepository.findOne({
+      where: { id: productId },
+      relations: ['org'],
+    });
+    if (!product) {
+      throw new NotFoundException({
+        status_code: 404,
+        message: 'Product not found',
+      });
+    }
+    if (product.org.id !== orgId) {
```

Comment:
If indeed it agrees with the requirements of the ownership guard, there should be no need for this if block